torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -3,19 +3,79 @@
 """
 from __future__ import annotations
 
+import functools
 import logging
-from typing import TYPE_CHECKING
+import os
 
-from torch._inductor.codecache import _ident, FxGraphCachePickler
+import torch
+from torch._inductor.codecache import (
+    _ident,
+    FxGraphCachePickler,
+    get_code_hash,
+    get_inductor_root,
+)
+from torch.fx.node import Node
 
 from .schemas import AOTConfig  # noqa: F401
 
-if TYPE_CHECKING:
-    import torch
-
 log = logging.getLogger(__name__)
 
 
+class BypassAOTAutogradCache(Exception):
+    pass
+
+
+def check_node_safe(node: Node):
+    """
+    Checks that the node only uses supported operators.
+    """
+
+    def is_torch_function(target):
+        return type(target).__name__.startswith("builtin")
+
+    def is_tensor(target):
+        # Tensors always have example values in meta field
+        return "example_value" in target.meta
+
+    # I'd love to use a match statement here, but it wasn't introduced until py3.10
+    if node.op == "call_function":
+        # We support only torch.* functions for now
+        # We can probably add an allowlist of safe non-torch implementations as well
+        if not is_torch_function(node.target):
+            raise BypassAOTAutogradCache(
+                f"Unsupported call_function target {node.target}"
+            )
+    elif node.op == "call_method":
+        method_target = node.args[0]
+        if not is_tensor(method_target):
```

Comment:
hmmm... if a user subclasses the `torch.Tensor` type, we probably can't guarantee that arbitrary (user-defined) tensor methods are cache-friendly.

From a quick check:
```
(Pdb) p type(torch.Tensor.cos)
<class 'method_descriptor'>
(Pdb) p type(torch.distributed._tensor.api.DTensor.redistribute)
<class 'function'>
```

In particular, `redistribute` is a tensor method defined on a subclass, but the way dynamo captures `DTensor.redistribute` is not currently cache-friendly, since dynamo closes over the `Placement` arguments. We want to fix this, but it seems safer to include that sort of thing in a separate allow-list than to blanket accept all tensor methods.

As a first / conservative approximation, checking that `type(target) is <class 'method_descriptor'>` seems reasonable, wdyt?

Reply:
Good idea, will do!

Reply:
Hmm so: when dynamo gives us an fx graph, the type of a "call_method" target seems to always be string. Some methods, such as DTensor.redistribute, actually get changed into functions:
```
 # code: return x.redistribute(mesh, [Replicate()]).to_local() + 2
prim_redistribute = torch__dynamo_variables_tensor_prim_redistribute(l_x_);  l_x_ = None
```
Others, like `DTensor.full_tensor()` dont' get converted, but the type of the `target` attr on the node is still just a string `"full_tensor"`. 

Will punt this part to a followup for now, but for the particular example, it looks like the conservative check above for function calls (i.e. that it's a builtin_function_or_method) already forbids it from being cached.

Reply:
ah right - sounds good (yep, forgot that dynamo will create a fresh function for that DTensor case)